### PR TITLE
`Paywalls`: finished `iOS 15` support

### PR DESCRIPTION
**Pull request by NachoSoto**

### CODE REVIEW

File: Tests/TestingApps/SimpleApp/SimpleApp/SimpleApp.swift [hunk 1]
```diff
@@ -27,9 +27,7 @@ struct SimpleApp: App {
 
     var body: some Scene {
         WindowGroup {
-            NavigationView {
```
Comment (NachoSoto) [inline Tests/TestingApps/SimpleApp/SimpleApp/SimpleApp.swift:30]: This was leading to a doubly-nested navigation view, which broke iOS 15.

File: RevenueCatUI/Templates/Template4View.swift [hunk 2]
```diff
@@ -107,8 +107,8 @@ struct Template4View: TemplateViewType {
 
     private var packagesScrollView: some View {
         self.packages
-            .frame(height: self.packageContentHeight)
             .scrollableIfNecessary(.horizontal)
+            .frame(height: self.packageContentHeight)
```
Comment (NachoSoto) [inline RevenueCatUI/Templates/Template4View.swift:111]: This should be equivalent, but only way to make it work in iOS 15.

File: RevenueCatUI/Templates/Template2View.swift [hunk 3]
```diff
@@ -38,6 +38,8 @@ struct Template2View: TemplateViewType {
     @ViewBuilder
     var content: some View {
         VStack(spacing: 10) {
+            Spacer()
```
Comment (NachoSoto) [inline RevenueCatUI/Templates/Template2View.swift:41]: This is the only change in this PR that affects iOS 16.

File: RevenueCatUI/Modifiers/CardHidingModifier.swift [hunk 4]
```diff
@@ -41,9 +41,10 @@ private struct CardHidingModifier: ViewModifier {
                 .padding(.vertical)
 
         case .condensedCard:
+            // "Hidden view" so it doesn't contribute to size calculation
             Rectangle()
-                // "Hidden view" so it doesn't contribute to size calculation
-                .frame(height: 0)
+                .frame(height: VersionDetector.iOS15 ? 1 : 0) // Note: height "0" breaks iOS 15
```
Comment (joshdholtz) [inline RevenueCatUI/Modifiers/CardHidingModifier.swift:46]: That is so weird 🤦‍♂️

**Review by joshdholtz — APPROVED** [review]
This looks good! 🙌

**Pull request merged**